shop/shop/src/test/java/ro/msg/learning/shop/services/strategies/SingleLocationStrategyTest.java

```diff
@@ -0,0 +1,131 @@
+package ro.msg.learning.shop.services.strategies;
+
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ro.msg.learning.shop.entities.*;
+import ro.msg.learning.shop.exceptions.MissingStockException;
+import ro.msg.learning.shop.services.LocationService;
+import ro.msg.learning.shop.services.StockService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+@ExtendWith(MockitoExtension.class)
+class SingleLocationStrategyTest {
+    private static final String COUNTRY = "ROMANIA";
+    private static final String CITY = "Sibiu";
+    private static final String COUNTY = "Sibiu";
+    private static final String STREET = "Calea Surii Mici";
+    private static final String CLUJ_CITY = "Cluj Napoca";
+    private static final String CLUJ_COUNTY = "Cluj";
+
+    private Location locationSibiu;
+    private Location locationCluj;
+    private ProductCategory productCategory;
+    private Supplier supplierEmag;
+    private Supplier supplierAltex;
+    private Product productRazerKeyboard;
+    private Product productDellLaptop;
+    private Stock stockRazerKeyboardCluj;
+    private Stock stockDellLaptopCluj;
+    private Stock stockRazerKeyboardSibiu;
+    private Stock stockDellLaptopSibiu;
+    private List<OrderDetail> orderDetails;
+    private List<Location> locations;
+
+    @Mock
+    private LocationService locationService;
+    @Mock
+    private StockService stockService;
+    @InjectMocks
+    private SingleLocationStrategy singleLocationStrategy;
+
+    void init() {
```

Comment:
Instead of calling this method before each test, you can annotated it with `@BeforeEach`. The behavior is the the same, but the code is a little bit cleaner :)

Same comment for the other test class